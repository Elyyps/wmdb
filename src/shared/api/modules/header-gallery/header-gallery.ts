export interface IHeaderGallery {
  id: number;
  type: "video" | "image";
  url: string;
}
