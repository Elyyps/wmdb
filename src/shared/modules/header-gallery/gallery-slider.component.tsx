import React from "react";
import Slider, { Settings } from "react-slick";
import { BasicArrow } from "./basic-arrow.component";
import { ImageComponent } from "@app/core/image";

export interface IGallerySliderComponentProps {
  currentIndex: number;
  images: string[];
}

const GallerySliderComponent = ({ currentIndex, images }: IGallerySliderComponentProps) => {
  const ref: any = React.createRef();

  const settings: Settings = {
    dots: false,
    arrows: true,
    infinite: true,
    prevArrow: <BasicArrow prev={true} />,
    nextArrow: <BasicArrow />,
    responsive: [
      {
        breakpoint: 1199,
        settings: {
          arrows: false,
          dots: true,
          slidesToScroll: 1,
          slidesToShow: 1
        }
      }
    ],
    slidesToScroll: 1,
    slidesToShow: 1,
    speed: 500,
    initialSlide: currentIndex
  };

  return (
    <Slider ref={ref} {...settings} initialSlide={currentIndex}>
      {images.map((item: string, key: number) => (
        <ImageComponent src={item} key={key} />
      ))}
    </Slider>
  );
};

export { GallerySliderComponent };
