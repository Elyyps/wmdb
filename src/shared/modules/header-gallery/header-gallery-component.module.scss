.header-gallery {
  &__container {
    display: flex;
    margin: 0 -32px;
    border-radius: $global-border-radius;
    overflow: hidden;
  }
  &__slider_images {
    img {
      height: 496px !important;
    }
  }
  &__slider_arrows {
    position: absolute;
    top: 50%;
    height: 50px;
    width: 100%;
    div {
      position: absolute;
      cursor: pointer;
      height: 48px;
      width: 48px;
      opacity: 1;
      -webkit-transform: translateY(-50%);
      transform: translateY(-50%);
      -webkit-transition: opacity 0.3s;
      transition: opacity 0.3s;
      background-color: rgba(0, 0, 0, 0.5);
      border-radius: 50%;
      border: none;
      span {
        margin: auto;
      }
    }
    &_prev {
      left: 10px;
    }
    &_next {
      right: 10px;
    }
  }

  &__col {
    box-sizing: border-box;
    &-card {
      img {
        height: 248px;
      }
    }
    &-cards {
      img {
        height: 116px;
      }
    }
    img {
      object-fit: cover;
      width: 100%;
    }
    &.large {
      width: 760px;
      position: relative;

      @media (max-width: $breakpoint-medium-max) {
        width: 100%;
      }
    }

    &.small {
      flex: 1;
    }
    .uk-grid-small {
      margin-left: 0;
      height: 496px;
      > * {
        padding-left: 8px;
      }

      > *:not(:first-child) {
        padding-top: 8px;
      }
    }
  }
  &__info {
    position: absolute;
    bottom: 24px;
    left: 24px;
    background-color: $global-inverse-color;
    box-shadow: 0 6px 12px 0 rgba($black, 0.06);
    padding: 12px 16px;
    border-radius: $global-border-radius;

    .slider-info {
      display: flex;
      align-items: center;
      margin: 0 -8px;
      > * {
        font-size: 14px;
        font-weight: 600;
        padding: 0 8px;
        display: flex;
        align-items: center;

        span {
          margin-right: 8px;
        }
      }
      .photo {
        color: $brand-primary;
        svg {
          path {
            fill: $brand-primary;
          }
        }
      }
    }
  }
}
