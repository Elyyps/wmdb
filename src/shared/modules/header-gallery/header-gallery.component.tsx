import * as React from "react";
import styles from "./header-gallery-component.module.scss";
import { IconComponent } from "@app/core/icon";
import { ImageComponent } from "@app/core/image";
import IconImage from "@assets/icons/images.svg";
import IconPlay from "@assets/icons/play.svg";
import Slider, { Settings } from "react-slick";
import { BasicArrow } from "./basic-arrow.component";

export interface IHeaderGalleryComponentProps {
  images: string[];
}

const HeaderGalleryComponent = (props: IHeaderGalleryComponentProps) => {
  const [currentImage, setCurrentImage] = React.useState(0);
  const smallImagesStart = 2;
  const handleChange = (prev: number, next: number) => {
    if (next === 1) {
      setCurrentImage(currentImage + 1);
    } else {
      setCurrentImage(currentImage - 1);
    }
  };

  React.useEffect(() => {
    currentImage >= props.images.length && setCurrentImage(0);
    currentImage < 0 && setCurrentImage(props.images.length - 1);
  }, [currentImage]);

  const ref: any = React.createRef();
  ref.current = "slick";

  const settings: Settings = {
    dots: false,
    arrows: true,
    infinite: true,
    prevArrow: <BasicArrow prev={true} />,
    nextArrow: <BasicArrow />,
    responsive: [
      {
        breakpoint: 1199,
        settings: {
          arrows: false,
          dots: true,
          slidesToScroll: 1,
          slidesToShow: 1
        }
      }
    ],
    slidesToScroll: 1,
    slidesToShow: 1,
    speed: 500
  };

  return (
    <div className={styles["header-gallery"]}>
      <div className="uk-container">
        <div className={styles["header-gallery__container"]}>
          <div className={`${styles["header-gallery__col"]} ${styles["large"]}`}>
            <Slider ref={ref} {...settings} beforeChange={handleChange}>
              {props.images.slice(currentImage, props.images.length).map((item: string, key: number) => (
                <ImageComponent src={item} key={key} />
              ))}
            </Slider>
            <div className={` ${styles["header-gallery__info"]} ${"uk-visible@l"} `}>
              <div className={styles["slider-info"]}>
                <div className={styles["photo"]}>
                  <IconComponent icon={IconImage} size={"15px"} />
                  Fotoâ€™s ({currentImage + 1}/{props.images.length})
                </div>
                <div className={styles["video"]}>
                  <IconComponent icon={IconPlay} size={"12px"} />
                  Video
                </div>
              </div>
            </div>
          </div>
          <div className={` ${styles["header-gallery__col"]} ${styles["small"]} ${"uk-visible@l"} `}>
            <div className={` uk-grid ${styles["uk-grid-small"]} data-uk-margin `}>
              {
                <>
                  <div className="uk-width-1-1" style={{ cursor: "pointer" }}>
                    <ImageComponent src={props.images[1]} onClick={() => setCurrentImage(1)} />
                  </div>
                  {props.images.slice(smallImagesStart, props.images.length).map((value: string, key: number) => (
                    <div className="uk-width-1-2@s" key={key} style={{ cursor: "pointer" }}>
                      <ImageComponent src={value} onClick={() => setCurrentImage(key + smallImagesStart)} />
                    </div>
                  ))}
                </>
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export { HeaderGalleryComponent };
