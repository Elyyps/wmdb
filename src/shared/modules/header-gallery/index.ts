export * from "./header-gallery.component";
export * from "./basic-arrow.component";
