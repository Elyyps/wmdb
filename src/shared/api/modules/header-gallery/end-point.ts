import { generateHeaderGalleryDummyData, generateGalleryDummyData } from "./generate-dummy-data";

export const generateHeaderGalleryData = generateHeaderGalleryDummyData;
export const generateGalleryData = generateGalleryDummyData;
