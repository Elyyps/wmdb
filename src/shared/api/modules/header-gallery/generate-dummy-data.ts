import Image1 from "@assets/img-1.jpg";
import Image2 from "@assets/img-2.png";
import Image3 from "@assets/img-3.png";
import Image4 from "@assets/img-4.png";
import Image5 from "@assets/img-5.png";
import Image6 from "@assets/img-6.png";
import { IHeaderGallery } from "./header-gallery";
export const generateHeaderGalleryDummyData = (): string[] => [
  Image1,
  Image2,
  Image3,
  Image4,
  Image5,
  Image6,
  Image2,
  Image3,
  Image4,
  Image5,
  Image6
];
export const generateGalleryDummyData = (): IHeaderGallery[] => [
  { type: "image", url: Image1, id: 0 },
  { type: "image", url: Image2, id: 1 },
  { type: "image", url: Image3, id: 2 },
  { type: "image", url: Image4, id: 3 },
  { type: "image", url: Image5, id: 4 },
  { type: "image", url: Image6, id: 5 },
  { type: "video", url: "https://www.youtube.com/watch?v=2oyhlad64-s", id: 6 },
  { type: "video", url: "https://www.youtube.com/watch?v=7gphiFVVtUI", id: 7 }
];
