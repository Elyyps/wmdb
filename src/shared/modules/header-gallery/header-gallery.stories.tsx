import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { HeaderGalleryComponent } from "./header-gallery.component";
import Oasis from "@assets/oasis.jpg";
import Image from "@assets/img02.png";
import Rectangle from "@assets/rectangle.png";
import Big from "@assets/img-big.jpg";
import Grid from "@assets/img-grid.png";
const imageList = [Oasis, Image, Oasis, Big, Grid, Rectangle];
storiesOf("HeaderGallery", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <HeaderGalleryComponent images={imageList} />);
