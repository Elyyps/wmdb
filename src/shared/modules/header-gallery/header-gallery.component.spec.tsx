import React from "react";
import { shallow } from "enzyme";
import { HeaderGalleryComponent } from "./header-gallery.component";
import Oasis from "@assets/oasis.jpg";
import Image from "@assets/img02.png";
import Rectangle from "@assets/rectangle.png";
import Big from "@assets/img-big.jpg";
import Grid from "@assets/img-grid.png";
const imageList = [Oasis, Image, Oasis, Big, Grid, Rectangle];
describe("[HeaderGallery]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<HeaderGalleryComponent images={imageList} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
