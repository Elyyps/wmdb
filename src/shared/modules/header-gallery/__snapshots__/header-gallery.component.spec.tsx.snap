// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HeaderGallery] should render component without crashing 1`] = `
<div
  className="header-gallery"
>
  <div
    className="uk-container"
  >
    <div
      className="header-gallery__container"
    >
      <div
        className="header-gallery__col large"
      >
        <Slider
          arrows={true}
          beforeChange={[Function]}
          dots={false}
          infinite={true}
          nextArrow={<BasicArrow />}
          prevArrow={
            <BasicArrow
              prev={true}
            />
          }
          responsive={
            Array [
              Object {
                "breakpoint": 1199,
                "settings": Object {
                  "arrows": false,
                  "dots": true,
                  "slidesToScroll": 1,
                  "slidesToShow": 1,
                },
              },
            ]
          }
          slidesToScroll={1}
          slidesToShow={1}
          speed={500}
        >
          <ImageComponent
            key="0"
            src="oasis.jpg"
          />
          <ImageComponent
            key="1"
            src="img02.png"
          />
          <ImageComponent
            key="2"
            src="oasis.jpg"
          />
          <ImageComponent
            key="3"
            src="img-big.jpg"
          />
          <ImageComponent
            key="4"
            src="img-grid.png"
          />
          <ImageComponent
            key="5"
            src="rectangle.png"
          />
        </Slider>
        <div
          className=" header-gallery__info uk-visible@l "
        >
          <div
            className="slider-info"
          >
            <div
              className="photo"
            >
              <IconComponent
                icon="images.svg"
                size="15px"
              />
              Fotoâ€™s (
              1
              /
              6
              )
            </div>
            <div
              className="video"
            >
              <IconComponent
                icon="play.svg"
                size="12px"
              />
              Video
            </div>
          </div>
        </div>
      </div>
      <div
        className=" header-gallery__col small uk-visible@l "
      >
        <div
          className=" uk-grid uk-grid-small data-uk-margin "
        >
          <div
            className="uk-width-1-1"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ImageComponent
              onClick={[Function]}
              src="img02.png"
            />
          </div>
          <div
            className="uk-width-1-2@s"
            key="0"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ImageComponent
              onClick={[Function]}
              src="oasis.jpg"
            />
          </div>
          <div
            className="uk-width-1-2@s"
            key="1"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ImageComponent
              onClick={[Function]}
              src="img-big.jpg"
            />
          </div>
          <div
            className="uk-width-1-2@s"
            key="2"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ImageComponent
              onClick={[Function]}
              src="img-grid.png"
            />
          </div>
          <div
            className="uk-width-1-2@s"
            key="3"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ImageComponent
              onClick={[Function]}
              src="rectangle.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
